src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -5,27 +5,117 @@ import language.implicitConversions
 import scala.compat.java8.collectionImpl._
 import scala.compat.java8.runtime._
 
-/** Classes or objects implementing this trait create streams suitable for sequential use */
-trait MakesSequentialStream[A, SS <: java.util.stream.BaseStream[A, SS]] extends Any {
-  def seqStream: SS
+trait MakesStepper[T, +Extra] extends Any {
+  /** Generates a fresh stepper of type `S` for element type `T` */
+  def stepper[S <: Stepper[_]](implicit ss: StepperShape[T, S]): S with Extra
 }
 
-/** Classes or objects implementing this trait create streams suitable for parallel use */
-trait MakesParallelStream[A, SS <: java.util.stream.BaseStream[A, SS]] extends Any {
-  def parStream: SS
+trait MakesKeyValueStepper[K, V, +Extra] extends Any {
+  /** Generates a fresh stepper of type `S` over map keys of type `K` */
+  def keyStepper[S <: Stepper[_]](implicit ss: StepperShape[K, S]): S with Extra
+
+  /** Generates a fresh stepper of type `S` over map values of type `V` */
+  def valueStepper[S <: Stepper[_]](implicit ss: StepperShape[V, S]): S with Extra
+}
+
+sealed trait StepperShape[T, S <: Stepper[_]] { def ref: Boolean }
+object StepperShape extends StepperShapeLowPrio {
+  private[this] def valueShape[T, S <: Stepper[_]]: StepperShape[T, S] = new StepperShape[T, S] { def ref = false }
+
+  // primitive
+  implicit val IntValue     = valueShape[Int, IntStepper]
+  implicit val LongValue    = valueShape[Long, LongStepper]
+  implicit val DoubleValue  = valueShape[Double, DoubleStepper]
+
+  // widening
+  implicit val ByteValue    = valueShape[Byte, IntStepper]
+  implicit val ShortValue   = valueShape[Short, IntStepper]
+  implicit val CharValue    = valueShape[Char, IntStepper]
+  implicit val FloatValue   = valueShape[Float, DoubleStepper]
+}
+trait StepperShapeLowPrio {
```

Comment:
Again, don't think it's worth the potential loss of clarity to save the characters `rity`.